### PR TITLE
bin/checkout: Port some to new style

### PR DESCRIPTION
**Pull request by cgwalters**
I was reading this code for unrelated reasons and noticed it
was still old style; port most (but not all) to new style.

### CODE REVIEW

File: src/ostree/ot-builtin-checkout.c [hunk 1]
```diff
@@ -145,45 +143,23 @@ process_one_checkout (OstreeRepo           *repo,
...
-          goto out;
-        }
+        return glnx_throw (error, "Cannot specify both --union and --union-identical");
       if (opt_union_add && opt_union_identical)
-        {
-          g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
-                       "Cannot specify both --union-add and --union-identical ");
-          goto out;
-        }
+        return glnx_throw (error, "Cannot specify both --union-add and --union-identical ");
```
Comment (jlebon) [inline src/ostree/ot-builtin-checkout.c:150]: Minor: looks like errant trailing space here, but it's not new.

File: src/ostree/ot-builtin-checkout.c [hunk 2]
```diff
@@ -198,11 +174,11 @@ process_one_checkout (OstreeRepo           *repo,
           if (!glnx_opendirat (AT_FDCWD, opt_selinux_policy, TRUE, &rootfs_dfd, error))
             {
               g_prefix_error (error, "selinux-policy: ");
-              goto out;
+              return FALSE;
```
Comment (jlebon) [inline src/ostree/ot-builtin-checkout.c:177]: `glnx_prefix_error`?

**Review by jlebon — APPROVED** [review]
Some optional comments, but LGTM as is too. Feel free to add `lgtm` label.

**Review by lucab — APPROVED** [review]
/lgtm

**Pull request merged**